docs/rfc/rfc-7.adoc

```diff
@@ -0,0 +1,288 @@
+in the worst case (`P = (95 choose 51) / (100 choose 51) = 0.0253280224`).
+In the worst case, it takes approximately 60 hours to sign 15 inputs.
+
+This attack slows down the sweeping schedule but given that the active wallet is
+not performing redemptions or moving funds, it does not affect the funds already
+bridged.
+
+In the first release, the client should assume no more than 3% of the network is
+corrupted and retry the signing of a single input no more than 10 times. If the
+signing of an input failed 10 times, the entire sweep is stopped. This puts the
+group in danger of a fraud challenge from one of the group members but the group
+will have a chance to sweep the deposit during the next sweep window. To defeat
+the fraud challenge, it is enough that the deposit has been swept, and it does
+not matter in which particular transaction.
+
+In the first release, the group maintains the signing state in memory, without
+persisting it to disk. It means that if more than t+1 members restarted their
+clients, the current sweep will not succeed, and everyone needs to wait for the
+next sweep window.
+
+==== Increasing fee
+
+At any point in time, the current Coordinator (who is not necessarily the same
+Coordinator who proposed the batch) can propose increasing the Bitcoin fee for
+the sweep transaction that is in the mempool. The signing group members sign the
+transaction if the original transaction is in the mempool for at least
+`sweeping_fee_bump_period` and the fee increase does not exceed the maximum one
+allowed by the Bridge. Given that the wallet is blocked with any other action
+until the sweep transaction is in the mempool, the fee bump signing protocol
+retries until the signature is produced or until the transaction is mined.
+
+When increasing a fee, we increment a fee multiplier:
+`fee_multiplier = fee_multiplier + sweeping_fee_multiplier_increment` and then
+calculate the new fee: `fee = base_fee * fee_multiplier`. We repeat until
+either the transaction posts or the fee exceeds the maximum one allowed by the
+Bridge.
+
+The wallet members must hold all the signed preimages to be able to defeat fraud
+challenges, even if the signatures were not used for the eventually mined
+transaction, that is, the transaction fee was increased.
+
+==== Governable parameters
+
+- `sweep_period`: The amount of time we wait between scheduled sweeps on a wallet.
+- `sweep_max_deposits`: The number of non-dust unswept revealed bitcoin deposits
+  that will trigger an early sweep on a wallet.
+- `sweeping_fee_bump_period`: The amount of time we wait to see if a sweeping
+  transaction is mined before increasing the fee.
+- `sweeping_fee_multiplier_increment`: The amount we add to the sweeping fee
+  multiplier each time a sweeping transaction is not mined within the
+  `sweeping_fee_bump_period`. For example, if this param is set to 0.2 and we
+  are currently at 1.6x, then the next time we would try 1.8x.
```

Comment:
Is it intentional that we don't mention `sweeping_refund_safety_time`, `dust_threshold`, `base_btc_fee_max`, `sweeping_fee_max_multiplier` and `btc_fee_max` here? I know we don't mention them in the document explicitly, but we refer to some of them indirectly (for example when mentioning `non-dust unswept revealed bitcoin deposits` or `the fee exceeds the maximum one allowed by the Bridge`).

Reply:
I added `sweeping_refund_safety_time` to the list and explicitly mentioned initial values for the parameters, see 45406ccdc677c9922bbe23ecf6bef3db937d840b. Regarding the rest of the parameters, they are already included in the `Bridge` contract... under different names that matched better the naming scheme we used. I did not include them to not cause confusion and just described them in English.